textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/JoinPredicate.java

```diff
@@ -0,0 +1,167 @@
+package edu.uci.ics.textdb.dataflow.common;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IPredicate;
+
+/**
+ * 
+ * @author sripadks
+ *
+ */
+public class JoinPredicate implements IPredicate {
+	
+	private Attribute outerIdAttribute;
+	private Attribute outerJoinAttribute;
+	private Attribute innerIdAttribute;
+	private Attribute innerJoinAttribute;
+	private Integer threshold;
+	
+	/**
+	 * <p>This constructor is used to set the parameters required for the Join Operator.</p>
+	 * 
+	 * <p>
+	 * JoinPredicate joinPre = new JoinPredicate(Attribute idAttr, Attribute descriptionAttr, Attribute idAttr, Attribute descriptionAttr, 10)
+	 * <br>will create a predicate that compares the spans of type descriptionAttr of outer and inner operators
+	 * (that agree on the idAttr id attributes) and outputs tuples which satisfy the criteria of being
+	 * within 10 characters of each other. </p>
+	 * 
+	 * <p>
+	 * Given below is a setting and an example using this setting to use JoinPredicate (consider the two 
+	 * tuples to be from two different operators).</p>
+	 * 
+	 * <p>
+	 * public static final Attribute idAttr = new Attribute("id", FieldType.INTEGER);
+	 * <br>public static final Attribute authorAttr = new Attribute("author", FieldType.STRING);
+	 * <br>public static final Attribute titleAttr = new Attribute("title", FieldType.STRING);
+	 * <br>public static final Attribute pagesAttr = new Attribute("numberOfPages", FieldType.INTEGER);
+	 * <br>public static final Attribute reviewAttr = new Attribute("reviewOfBook", FieldType.TEXT);
+	 * </p>
+	 * <p>
+	 * public static final Attribute[] bookAttr = { idAttr, authorAttr, titleAttr, pagesAttr, reviewAttr, 
```

Comment:
To increase its readability, please convert the code-based example to a text-based example.  Also indicate the output of the operator (especially the start/end of the generated span).